theano/gof/cmodule.py

```diff
@@ -2004,6 +2026,21 @@ def join_options(init_part):
         if GCC_compiler.march_flags:
             cxxflags.extend(GCC_compiler.march_flags)
 
+        # Find working march flag:
+        verbose = True
+        march_flags_to_try = [cxxflags[0], '-march=core2', '-march=corei7', '-march=corei7-avx']
```

Comment:
Reverse the order of the strings we try. It should be done earlier as wrote before as we also want to cache the try of the flags.